src-ghc/Pact/Interpreter.hs

```diff
@@ -298,16 +298,23 @@ interpret :: Interpreter e -> EvalEnv e -> EvalInput -> IO EvalResult
 interpret runner evalEnv terms = do
   ((rs,logs,txid),state) <-
     runEval def evalEnv $ evalTerms runner terms
-  gas <- readIORef (_eeGas evalEnv)
+  milliGas <- readIORef (_eeGas evalEnv)
   warnings <- readIORef (_eeWarnings evalEnv)
-  let gasLogs = _evalLogGas state
+  let pact48Disabled = views (eeExecutionConfig . ecFlags) (S.member FlagDisablePact48) evalEnv
+      gasLogs = _evalLogGas state
       pactExec = _evalPactExec state
       modules = _rsLoadedModules $ _evalRefs state
+      gasUsed = if pact48Disabled then milliGasToGas milliGas else gasRem milliGas
   -- output uses lenient conversion
   return $! EvalResult
     terms
     (map (elideModRefInfo . toPactValueLenient) rs)
-    logs pactExec gas modules txid gasLogs (_evalEvents state) warnings
+    logs pactExec gasUsed modules txid gasLogs (_evalEvents state) warnings
+  where
+    -- Round up by 1 if the `MilliGas` amount is in any way fractional.
+    gasRem (MilliGas milliGas) =
+      let (d, r) = milliGas `divMod` millisPerGas
```

Comment:
`quotRem` here is a small improvement for something we're gonna be caculating constantly.

Reply:
👍 true, and this amount is never negative. Will change.